packages/customer-account-ui-extensions/src/extension-points/standard-api.ts

```diff
@@ -13,11 +14,115 @@ export interface Language {
   isoCode: string;
 }
 
+export interface Currency {
+  /**
+   * The ISO-4217 code for this currency.
+   * @see https://www.iso.org/iso-4217-currency-codes.html
+   */
+  isoCode: CurrencyCode;
+}
+
 export interface Localization {
   /**
-   * The language the buyer sees in the customer account hub.
+   * The currency that the buyer sees for money amounts in the checkout.
+   */
+  currency: StatefulRemoteSubscribable<Currency>;
+
+  /**
+   * The buyer’s time zone.
+   */
+  timezone: StatefulRemoteSubscribable<Timezone>;
+
+  /**
+   * The language the buyer sees in the checkout.
    */
   language: StatefulRemoteSubscribable<Language>;
+
+  /**
+   * This is the buyer's language, as supported by the extension.
+   * If the buyer's actual language is not supported by the extension,
+   * this is the fallback locale used for translations.
+   *
+   * For example, if the buyer's language is 'fr-CA' but your extension
+   * only supports translations for 'fr', then the `isoCode` for this
+   * language is 'fr'. If your extension does not provide french
+   * translations at all, this value is the default locale for your
+   * extension (that is, the one matching your .default.json file).
+   */
+  extensionLanguage: StatefulRemoteSubscribable<Language>;
+}
+
+/**
+ * This defines the i18n.translate() signature.
+ */
+export interface I18nTranslate {
+  /**
+   * This returns a translated string matching a key in a locale file.
+   *
+   * @example translate("banner.title")
+   */
+  <ReplacementType = string>(
+    key: string,
+    options?: {[placeholderKey: string]: ReplacementType | string | number},
+  ): ReplacementType extends string | number
+    ? string
+    : (string | ReplacementType)[];
+}
+
+export interface I18n {
+  /**
+   * Returns a localized number.
+   *
+   * This function behaves like the standard `Intl.NumberFormat()`
+   * with a style of `decimal` applied. It uses the buyer's locale by default.
+   *
+   * @param options.inExtensionLocale - if true, use the extension's locale
+   */
+  formatNumber: (
+    number: number | bigint,
+    options?: {inExtensionLocale?: boolean} & Intl.NumberFormatOptions,
+  ) => string;
+
+  /**
+   * Returns a localized currency value.
+   *
+   * This function behaves like the standard `Intl.NumberFormat()`
+   * with a style of `currency` applied. It uses the buyer's locale by default.
+   *
+   * @param options.inExtensionLocale - if true, use the extension's locale
+   */
+  formatCurrency: (
+    number: number | bigint,
+    options?: {inExtensionLocale?: boolean} & Intl.NumberFormatOptions,
+  ) => string;
```

Comment:
I wonder what currency should this method use? I understand that in Checkout the have a currency, but in Self-Serve a currency only exists in specific context, ie within an order.

I would need to see the implementation to understand a bit better whether this works for our use case.

Reply:
The function itself makes sense since a currency may be present within the data and an extension might want to render it.

Though I do not believe we have a default currency for the user right now, just like we do not have a different locale for the extension even though it could be added in the future.

Reply:
Update: I decided to not include currency in our API for now, as I did not have a default currency for the user right now. However, we can always add that in the future.

Thanks @emileber